### PR TITLE
Deprecate an entity

### PR DESCRIPTION
**Pull request by Ankit-Keshari-Vituity**
Added a button to "deprecate" an entity from inside of the UI from the following entity profile page: dataset, chart, dashboard, data flow, data job, container, glossary term

deprecate_entity

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/entity/shared/containers/profile/header/AddDeprecatedDataModal.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                message.error({ content: `Failed to update Deprecation: \n ${e.message || ''}`, duration: 2 });
+            }
+        }
+        refetch?.();
+        handleClose();
+    };
+
+    return (
+        <Modal
+            title="Add Deprecation Data"
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/header/AddDeprecatedDataModal.tsx:48]: Minor: Let's change to "Deprecation Details"

File: datahub-web-react/src/app/entity/shared/containers/profile/header/AddDeprecatedDataModal.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import { useUpdateDeprecationMutation } from '../../../../../../graphql/mutations.generated';
+
+type Props = {
+    urn: string;
+    visible: boolean;
+    onClose: () => void;
+    refetch?: () => Promise<any>;
+};
+
+export const AddDeprecatedDataModal = ({ urn, visible, onClose, refetch }: Props) => {
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/header/AddDeprecatedDataModal.tsx:12]: Minor: Let's change name to "DeprecationDetailsModal"

File: datahub-web-react/src/app/entity/shared/containers/profile/header/AddDeprecatedDataModal.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            visible={visible}
+            onCancel={handleClose}
+            keyboard
+            footer={
+                <>
+                    <Button onClick={handleClose} type="text">
+                        Cancel
+                    </Button>
+                    <Button form="addDeprecationForm" key="submit" htmlType="submit">
+                        Add
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/header/AddDeprecatedDataModal.tsx:58]: Minor: let's rename this to "Ok"

File: datahub-web-react/src/app/entity/shared/containers/profile/header/AddDeprecatedDataModal.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+export const AddDeprecatedDataModal = ({ urn, visible, onClose, refetch }: Props) => {
+    const [updateDeprecation] = useUpdateDeprecationMutation();
+    const [form] = Form.useForm();
+
+    const handleClose = () => {
+        form.resetFields();
+        onClose();
+    };
+
+    const handleAdd = async (formData: any) => {
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/header/AddDeprecatedDataModal.tsx:21]: minor: let's rename to handleOk

File: datahub-web-react/src/app/entity/shared/containers/profile/header/AddDeprecatedDataModal.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                        Add
+                    </Button>
+                </>
+            }
+        >
+            <Form form={form} name="addDeprecationForm" onFinish={handleAdd} layout="vertical">
+                <Form.Item name="note" label="Note">
+                    <Input placeholder="Add Note" autoFocus />
+                </Form.Item>
+                <Form.Item name="decommissionTime" label="Decommission Time">
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/header/AddDeprecatedDataModal.tsx:67]: Minor: Decomission Date

File: datahub-web-react/src/app/entity/shared/containers/profile/header/EntityHeader.tsx [hunk 6]
```diff
@@ -120,68 +171,160 @@ export const EntityHeader = () => {
...
-                <Button
-                    icon={copiedUrn ? <CheckOutlined /> : <CopyOutlined />}
-                    onClick={() => {
-                        navigator.clipboard.writeText(urn);
-                        setCopiedUrn(true);
-                    }}
-                />
-            </Tooltip>
-        </HeaderContainer>
+                        {entityData?.deprecation?.deprecated && entityData?.deprecation?.decommissionTime !== null && (
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/header/EntityHeader.tsx:269]: Minor: Let's put this condition in a variable above^^ (line 227) that is called "isDeprecated" --> I think we can also remove the && `entityData?.deprecation?.decommissionTime !== null`

File: datahub-web-react/src/app/entity/shared/containers/profile/header/AddDeprecatedDataModal.tsx (hunk 1 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/entity/shared/containers/profile/header/AddDeprecatedDataModal.tsx]: Done and Committed.

File: datahub-web-react/src/app/entity/shared/containers/profile/header/AddDeprecatedDataModal.tsx (hunk 2 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/entity/shared/containers/profile/header/AddDeprecatedDataModal.tsx]: Done and Committed.

File: datahub-web-react/src/app/entity/shared/containers/profile/header/AddDeprecatedDataModal.tsx (hunk 3 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/entity/shared/containers/profile/header/AddDeprecatedDataModal.tsx]: Done and Committed.

File: datahub-web-react/src/app/entity/shared/containers/profile/header/AddDeprecatedDataModal.tsx (hunk 4 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/entity/shared/containers/profile/header/AddDeprecatedDataModal.tsx]: Done and Committed.

File: datahub-web-react/src/app/entity/shared/containers/profile/header/AddDeprecatedDataModal.tsx (hunk 5 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/entity/shared/containers/profile/header/AddDeprecatedDataModal.tsx]: Done and Committed.

File: datahub-web-react/src/app/entity/shared/containers/profile/header/EntityHeader.tsx (hunk 6 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/entity/shared/containers/profile/header/EntityHeader.tsx]: `{entityData?.deprecation?.deprecated &&
                            (entityData?.deprecation?.note !== '' ||
                                entityData?.deprecation?.decommissionTime !== null) && (
                                <Popover
                                    overlayStyle={{ maxWidth: 240 }}
                                    placement="right"
                                    content={
                                        <>
                                            {entityData?.deprecation?.note !== '' && (
                                                <Typography.Text>{entityData?.deprecation?.note}</Typography.Text>
                                            )}
                                            {entityData?.deprecation?.decommissionTime !== null && (
                                                <Typography.Text type="secondary">
                                                    <Tooltip placement="right" title={decommissionTimeGMT}>
                                                        <LastEvaluatedAtLabel
                                                            style={{
                                                                borderTop: '1px solid #f0f0f0',
                                                                paddingTop: '5px',
                                                            }}
                                                        >
                                                            {decommissionTimeLocal}
                                                        </LastEvaluatedAtLabel>
                                                    </Tooltip>
                                                </Typography.Text>
                                            )}
                                        </>
                                    }
                                >
                                    <DeprecatedContainer>
                                        <InfoCircleOutlined />
                                        <DeprecatedText>Deprecated</DeprecatedText>
                                    </DeprecatedContainer>
                                </Popover>
                            )}
                        {entityData?.deprecation?.deprecated &&
                            entityData?.deprecation?.decommissionTime === null &&
                            entityData?.deprecation?.note === '' && (
                                <DeprecatedContainer>
                                    <InfoCircleOutlined />
                                    <DeprecatedText>Deprecated</DeprecatedText>
                                </DeprecatedContainer>
                            )}
`

File: datahub-web-react/src/app/entity/shared/containers/profile/header/EntityHeader.tsx [hunk 7]
```diff
@@ -120,68 +171,177 @@ export const EntityHeader = () => {
...
+                                    <DeprecatedContainer>
+                                        <InfoCircleOutlined />
+                                        <DeprecatedText>Deprecated</DeprecatedText>
+                                    </DeprecatedContainer>
+                                </Popover>
+                            )}
+                        {entityData?.deprecation?.deprecated &&
+                            entityData?.deprecation?.decommissionTime === null &&
+                            entityData?.deprecation?.note === '' && (
+                                <DeprecatedContainer>
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/header/EntityHeader.tsx:306]: This is still duplicated

File: datahub-web-react/src/app/entity/shared/containers/profile/header/EntityHeader.tsx [hunk 8]
```diff
@@ -120,68 +171,177 @@ export const EntityHeader = () => {
...
-                <Button
-                    icon={copiedUrn ? <CheckOutlined /> : <CopyOutlined />}
-                    onClick={() => {
-                        navigator.clipboard.writeText(urn);
-                        setCopiedUrn(true);
-                    }}
-                />
-            </Tooltip>
-        </HeaderContainer>
+                        {entityData?.deprecation?.deprecated &&
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/header/EntityHeader.tsx:269]: lets do this. above create a new variable

const hasDetails = (entityData?.deprecation?.note !== '' ||
                                entityData?.deprecation?.decommissionTime !== null)
 
then inside the content box we can do this:

!hasDetails && <current line 276-295> || 'No additional details' 

then you can get rid of the long conditional on 269, as well as the line 303-310


entityData?.deprecation?.deprecated

Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/entity/shared/containers/profile/header/EntityHeader.tsx]: Fixed and committed my changes

**Review by jjoyce0510 — APPROVED** [review]

**Pull request merged**